recipe/meta.yaml

```diff
@@ -6,36 +6,59 @@ package:
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/gravis-{{ version }}.tar.gz
-  sha256: a36342602aa7da3bbd674c970439717926a8f7ca0678aa82cc5d756e8c2a529b
+  # switch to GH release archive, because of tests
+  url: https://github.com/robert-haas/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 8734066afd749fd85f9f173f92a1b501ae8e31723d1894c15f29fcec3eac5419
+  patches:
+    - patches/0001_fix-win-unicode-encoding.patch  # [win]
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
   number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - m2-patch    # [win]
   host:
-    - python >=3.5
+    - python
+    - setuptools
     - pip
+    - wheel
   run:
-    - python >=3.5
+    - python
+    # access to package files
     - setuptools >=40.0
 
 test:
   imports:
     - gravis
-  commands:
-    - pip check
+  source_files:
+    - tests
   requires:
     - pip
+    - pytest
+    - networkx
+    - ipython
```

Comment:
I don't think `ipython` is necessary.

Reply:
it is used in a test
https://github.com/robert-haas/gravis/blob/3e63b606141bf78617b1247048a9e3ccca6d5824/tests/test_plotting_options.py#L31

it's in an `_internal` folder
https://github.com/robert-haas/gravis/blob/3e63b606141bf78617b1247048a9e3ccca6d5824/gravis/_internal/plotting/data_structures.py#L40

either it is needed also for `run`, or it is intended as optional (with `fig.display(inline=True)`)

it is mentioned only in the tox.ini as test dep I supposed:
https://github.com/robert-haas/gravis/blob/3e63b606141bf78617b1247048a9e3ccca6d5824/tox.ini#L13

Reply:
@JeanChristopheMorinPerso could you please have a look at this? Is `ipython` optional as part of Jupiter Notebooks?

Reply:
I have no idea.

Reply:
Thank you for having had a look at it. 

resolution: I will keep it in the `test` section to make those some ones pass, and assume `ipython` is an optional dependency because it is not mentioned anywhere, and it is used only in certain cases (with `fig.display(inline=True)`), and the `ModuleNotFoundError: No module named 'IPython'` would be thrown: either the user will know to install `ipython` either will complain to us.

Reply:
FYI @ifitchet since we worked on this together